### PR TITLE
ID16.2 Connect the legend for live updates from backend data (Jira #ID-61)

### PR DESCRIPTION
**Pull request by Stebis-dev**
Connect the legend for live updates from backend data (Jira #ID-61)

### CODE REVIEW

File: src/app/services/stock-data-shared.service.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  private hoveredDataSubject = new BehaviorSubject<uPlot.AlignedData>([]);
+  hoveredData$ = this.hoveredDataSubject.asObservable();
+  private hoveredDataPercentage = new BehaviorSubject<
+    number | null | undefined
+  >(-1);
+  stockDataPercentage$ = this.hoveredDataPercentage.asObservable();
+
+  constructor() {}
+
+  // updateHoveredData(data: any[], stockPercentage: number[]): void {
```
Comment (winqus) [inline src/app/services/stock-data-shared.service.ts:19]: remove commented out code

File: src/app/components/stock-graph/stock-graph.component.ts [hunk 2]
```diff
@@ -115,30 +123,45 @@ export class StockGraphComponent {
 
     if (this.uPlotInstance) {
       this.uPlotInstance.setData(this.data);
+      this.sharedService.updateData(this.data);
+
+      // update data observable
```
Comment (winqus) [inline src/app/components/stock-graph/stock-graph.component.ts:128]: remove comments

**Review by winqus — CHANGES_REQUESTED** [review]

File: src/app/services/stock-data-shared.service.ts (hunk 1 above)
Comment (Stebis-dev) [reply src/app/services/stock-data-shared.service.ts]: fixed

File: src/app/components/stock-graph/stock-graph.component.ts (hunk 2 above)
Comment (Stebis-dev) [reply src/app/components/stock-graph/stock-graph.component.ts]: fixed

**Review by winqus — APPROVED** [review]

**Pull request merged**